File: questions/1/c++/1.cpp
#include <vector>
#include <map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int> nums, int target) {
        std::map<int, int> prevNums;

        for (int idx = 0; idx < nums.size(); ++idx) {
            int newTarget = target - nums[idx];

            if (prevNums.find(newTarget) != prevNums.end()) {
                return {prevNums[newTarget], idx};
            }

            prevNums[nums[idx]] = idx;
        }

        return {-1, -1};
    }
};

File: questions/1/csharp/1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var prevNums = new Dictionary<int, int>();

        for (var idx = 0; idx < nums.Length; ++idx) {
            var newTarget = target - nums[idx];

            if (prevNums.TryGetValue(newTarget, out int prevIdx)) {
                return new[] {prevIdx, idx};
            }

            prevNums[nums[idx]] = idx;
        }

        return new[] {-1, -1};
    }
}